autosubmit/helpers/processes.py

```diff
@@ -130,9 +134,6 @@ def _match_autosubmit_cmdline(cmdline: List[str], command='run', expid: Optional
     if filtered_list[1] != command:
         return None
 
-    if not _match_expid(filtered_list[2]):
-        return None
-
```

Comment:
This line was never reachable, as above we verify that the first argument is `autosubmit`, the second is `run`, and before we check that there are three elements in the list... and in the `filtered_list` creation we only have three elements being filtered, `autosubmit`, `run` (or another command) and the expid. Thus, we already know here that the last element in the array can only be an expid.